packages/medusa/src/subscribers/search-indexing.ts

```diff
@@ -74,16 +104,7 @@ class SearchIndexingSubscriber {
           "created_at",
           "updated_at",
         ],
-        relations: [
-          "variants",
-          "tags",
-          "type",
-          "collection",
-          "variants.prices",
-          "images",
-          "variants.options",
-          "options",
-        ],
+        relations,
         take: take,
```

Comment:
todo: depending, if we have a take, shouldn't we have a skip?

Reply:
The document indexer has its own skip-mechanism using the last seen id:
```
let lastSeenId = ""

while (hasMore) {
  const products = await this.retrieveNextProducts(lastSeenId, TAKE)

  if (products.length > 0) {
    await this.searchService_.addDocuments(
      ProductService.IndexName,
      products,
      indexTypes.products
    )
    lastSeenId = products[products.length - 1].id
  } else {
    hasMore = false
  }
}
```

Reply:
I am not sure that it works, last seen id is a string and we use the GT find operator. Have you tested it?

Reply:
I haven't tested this as it is not new functionality. It has always been like this.

That said, I'll make sure to run a test and fix, if this is not working as expected :)

Reply:
Also, since there is no where and that we are using skip/take, I would suggest to improve the perf by doing a first select of the product only without relations using limit/offset, then a select with all relations etc where id in the result of the previous one. It would avoid having the entire products and relations in memory before typeorm does the skip/take locally. wdyt?

Reply:
Pagination using the ulid works as expect 👍